src/v/bytes/iobuf.cc

```diff
@@ -200,3 +200,22 @@ bool iobuf::operator==(const iobuf& o) const {
     }
     return true;
 }
+
+bool iobuf::operator==(std::string_view o) const {
```

Comment:
@dotnwat I think you commented on this on my other PR. The idea is to be able to do something like:
```
static constexpr std::string_view hdr("header");
if(buf == hdr)
```
And this was my solution. Maybe theres a more elegant way?